MapboxCoreNavigation/RouteController.swift

```diff
@@ -530,21 +530,21 @@ extension RouteController: CLLocationManagerDelegate {
     }
 
     @objc public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        guard let location = locations.last else {
-            return
-        }
-        self.rawLocation = location
-
-        sessionState.pastLocations.push(location)
-
-        if location.isQualified {
+        let filteredLocations = locations.filter { $0.isQualified }
+        
+        if !filteredLocations.isEmpty, hasFoundOneQualifiedLocation == false {
```

Comment:
Nit: this property name is a mouthful, how about `hasQualifiedLocation`?

Reply:
`hasQualifiedLocation` implies something else. This variable is used really only at the beginning of a session when locations are generally bad. We don't emit that we discarded a location until we get at least one good location.